File: 백준/Silver/10431. 줄세우기/줄세우기.py
import sys
input = sys.stdin.readline

P = int(input())

for _ in range(P):
    T = list(map(int, input().split()))
    line = [T[1]]
    cnt = 0
    for i in range(2, len(T)):
        if max(line) > T[i]:
            for j in range(len(line)):
                if line[j] > T[i]:
                    cnt += len(line) - j
                    line = line[:j] + [T[i]] + line[j:]
                    break
        else:
            line.append(T[i])

    print(T[0], cnt)

File: 백준/Silver/10431. 줄세우기/줄세우기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	int P = Integer.parseInt(br.readLine());
    	
    	for (int p = 0; p < P; p++) {
    		StringTokenizer st = new StringTokenizer(br.readLine());
    		int T = Integer.parseInt(st.nextToken());
    		
    		List<Integer> line = new ArrayList<>(20);
    		int cnt = 0;
    		
    		for (int i = 0; i < 20; i++) {
    			int student = Integer.parseInt(st.nextToken());
    			int back = line.size();
    			
    			for (int j = 0; j < line.size(); j++) {
    				if (line.get(j) > student) {
    					back = j;
    					break;
    				}
    			}
    			
    			cnt += line.size() - back;
    			line.add(back, student);
    		}

    		System.out.println(T + " " + cnt);
    	}
    }
}